channel_layout.go

```diff
@@ -123,3 +130,28 @@ func (l ChannelLayout) clone() (ChannelLayout, error) {
 	dst := newChannelLayoutFromC(&cl)
 	return dst, err
 }
+
+// https://ffmpeg.org/doxygen/7.0/group__lavu__audio__channels.html#ga16a894cd275eba471ccad4fe1c5e5c28
+func ChannelLayoutFromString(s string) (ChannelLayout, error) {
+	var c C.AVChannelLayout
+	cn := C.CString(s)
+	defer C.free(unsafe.Pointer(cn))
+	if err := newError(C.av_channel_layout_from_string(&c, cn)); err != nil {
+		return ChannelLayout{}, fmt.Errorf("astiav: could not find or parse channel layout %q: %w", s, err)
+	}
+	return newChannelLayoutFromC(&c), nil
+}
+
+func (l ChannelLayout) MarshalText() ([]byte, error) { return ([]byte)(l.String()), nil }
+func (l *ChannelLayout) UnmarshalText(d []byte) error {
+	s := string(d)
+	if s == "" {
+		return nil
+	}
+	pf, err := ChannelLayoutFromString(s)
+	if err != nil {
+		return err
```

Comment:
Could you return this instead `fmt.Errorf("astiav: finding channel layout failed: %w", err)`?